File: C++/First Steps in Coding - Basics of C++/Conditional Statements - Advanced/Exercises/Ski Trip.cpp
#include <iostream>
#include <cstdlib>
#include <iomanip>
#include <string>
using namespace std;
int main() {
	int daysToStay;
	string roomType, rating;
	double cost=0;
	cin >> daysToStay;
	getline(cin>>ws,roomType); getline(cin, rating);
	cin >> rating;
	daysToStay -= 1;
	if (roomType == "room for one person") {
		cost += 18.00*daysToStay;
	}
	else if (roomType == "apartment") {
		cost += 25.00*daysToStay;
		if (daysToStay < 10) {
			cost -= (cost*0.30);
		}
		else if (daysToStay >= 10 and daysToStay <= 15) {
			cost -= (cost*0.35);
		}
		else if (daysToStay > 15) {
			cost -= (cost*0.50);
		}
	}
	else if (roomType == "president apartment") {
		cost += 35.00*daysToStay;
		if (daysToStay < 10) {
			cost -= (cost*0.10);
		}
		else if (daysToStay >= 10 and daysToStay <= 15) {
			cost -= (cost*0.15);
		}
		else if (daysToStay > 15) {
			cost -= (cost*0.20);
		}
	}

	if (rating == "positive") {
		cost += (cost*0.25);
	}
	else if (rating == "negative") {
		cost -= (cost*0.10);
	}
	cout << fixed << setprecision(2) << cost;
}

File: Python/Basics/Conditional Statements Advanced/Exercise/SkiTrip.py
days = int(input()) - 1
typeOfStay = input() #"room for one person", "apartment" или "president apartment"
estimate = input() # positive negative
total = 0
# 	"room for one person" – 18.00 лв за нощувка
# 	"apartment" – 25.00 лв за нощувка
# 	"president apartment" – 35.00 лв за нощувка

oneRoom = 18
apartment = 25
presidentAp = 35

if typeOfStay == "room for one person":
    total = days * oneRoom
elif typeOfStay == "apartment":
    total = days * apartment
elif typeOfStay == "president apartment":
    total = days * presidentAp

if typeOfStay == "apartment":
    if days < 10:
        total -= total * 0.3
    elif 10 <= days <= 15:
        total -= total * 0.35
    elif days > 15:
        total -= total * 0.5
elif typeOfStay == "president apartment":
    if days < 10:
        total -= total * 0.1
    elif 10 <= days <= 15:
        total -= total * 0.15
    elif days > 15:
        total -= total * 0.2


if estimate == "positive":
    total += total * 0.25
elif estimate == "negative":
    total -= total * 0.1

print(f"{total:.2f}")


